File: app/src/main/java/mishka/rssreader/di/component/PostComponent.java
package mishka.rssreader.di.component;

import dagger.Component;
import mishka.rssreader.di.module.ContextModule;
import mishka.rssreader.di.module.PostModule;
import mishka.rssreader.ui.post.PostActivity;

@Component(modules = {PostModule.class, ContextModule.class})
public interface PostComponent {
    void inject(PostActivity postActivity);
}

File: app/src/main/java/mishka/rssreader/di/component/PostComponent.kt
package mishka.rssreader.di.component

import dagger.Component
import mishka.rssreader.di.ActivityScope
import mishka.rssreader.di.module.PostModule
import mishka.rssreader.ui.post.PostFragment

@Component(modules = [PostModule::class], dependencies = [ApplicationComponent::class])
@ActivityScope
public interface PostComponent {
    fun inject(postFragment: PostFragment)
}
